src/data/fin/basic.lean

```diff
@@ -1234,26 +1234,38 @@ begin
   rwa subtype.ext_iff at h
 end
 
+lemma coe_sub_of_le {n : ℕ} {a b : fin n} (h : b ≤ a) :
+  (↑(a - b) : ℕ) = a - b :=
```

Comment:
Is the iff version any easier to prove starting from the golfed proofs? I feel like the missing piece is `nat.mod_eq_iff_lt` which I couldn't find

Reply:
Does proving `nat.mod_eq_iff_lt` make this proof easier? Or is it not true in any meaningful way?

Reply:
I pushed them, but I don't see how the iff version makes the `rw` we have simpler.